contributors/devel/sig-architecture/object-lifecycle.md

```diff
@@ -0,0 +1,94 @@
+* `ACTIVE`: The API server is aware of the object and the object has not been
+  deleted (the `metadata.deletionTimestamp` is not set).  While in this state,
+  any update operations (PUT, PATCH, server-side apply, etc.) will result in
+  this same state.
+* `DELETING`: The API server is aware of the object, and the object has been
+  deleted (the `metadata.deletionTimestamp` is set), and either the object has
+  one or more finalizers (in `metadata.finalizers`) or has a deletion grace
+  period (in `metadata.deletionGracePeriodSeconds`) greater than zero (NOTE:
+  most API types do not allow setting `deletionGracePeriodSeconds` at all).
+  When the last finalizer and/or grace period are removed, the object will
+  cease to exist.
+
+### State Machine Diagram
+
+The following diagram describes the above states:
+
+```
+                                  +---- object
+                                  |     updated
+                                  v        |
+                           +----------+    |
+                           |          +----+
+          object --------->|  ACTIVE  |
+          created          |          +-----------+
+             |             +---+------+           |
+             |                 |                  |
+             |                 |                  |
++------------+---+     object deleted             |
+|                |     without finalizers         |
+|                |<--- or grace period            |
+|                |                           object deleted
+| DOES_NOT_EXIST |                           with finalizers
+|                |                           or grace period
+|                |<--- finalizers removed         |
+|                |     and grace period           |
++----------------+     complete                   |
+                               |                  |
+                               |                  |
+                           +---+------+           |
+                           |          |           |
+                           | DELETING |<----------+
+                           |          |
+                           +----------+
+```
+
+### Handling DELETING
+
+The `DELETING` state warrants some special attention from controllers. Because
+this state did not really exist (for any API except Pods) until finalizers were
+added, many older controllers do not consider it at all.  Even newly written
+controllers do not always give it due consideration.
+
+When crafting a controller, authors must consider what semantics to apply to
+objects in the DELETING state.  Even if this state is not intended to be
+meaningful to a given API, it can be expressed and can not generally be
+prevented without significant effort.  It should therefore be considered and
+handled, or at least documented.
+
+To a controller, the transition from `ACTIVE` to `DELETING` appears as an
+update operation which sets the `metadata.deletionTimestamp` field.
```

Comment:
Can we add an example?
this is not supposed to be handled by adding a hook on the event handler directly, right?

Reply:
There is not a different hook, no.  It's just the update handler.

Reply:
That's what I meant, should not be managed in the sync loop and not in the update handlers?